rudderlabs/rudder-sdk-js
# chore(monorepo): update npm package dependencies

## PR Description

- Update dependencies to latest versions
- Cleanup dependency overrides
- Move linting to relevant nx executor
- Ensure NX Graph generates correct lib dependency tree
- add nx tags

## Linear task (optional)

Linear task link

## Cross Browser Tests

Please confirm you have tested for the following browsers:


## Security

- [ ] The code changed/added as part of this pull request won't create any security issues with how the software is being used.
